ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterManager.java

```diff
@@ -102,11 +114,16 @@
           @Override
           public void run() {
             try {
-              HelixManager manager =
-                  helixFactory.getZKHelixManager(clusterName, instanceName, InstanceType.SPECTATOR, zkConnectStr);
-              logger.info("Connecting to Helix manager at {}", zkConnectStr);
-              manager.connect();
-              logger.info("Established connection to Helix manager at {}", zkConnectStr);
+              HelixManager manager;
+              if (!dcName.equals(clusterMapConfig.clusterMapDatacenterName)) {
```

Comment:
I generally prefer `if (x) then y else z`  vs. `if (!x) then z else y`. Not a big deal here, but in a more general setting, you might end up doing double negation in your mind for the else case which can be annoying.

Reply:
sounds reasonable. Will take your advice.